OneSignal/OneSignal-Unity-SDK
# Fixed InstallEdm4uStep to properly work with UPM EDM4U installations

# Description
## Fixed InstallEdm4uStep to properly work with UPM EDM4U installations

## Details

### Motivation
EDM4U is a google unity package that is distributed in two versions: classic .unitypackage or .tgz via installing using Package Manager (https://developers.google.com/unity/archive). In our projects for the sake of keeping Assets folder as clean as possible we use second option. Right now InstallEdm4uStep fails to detect EDM4U version if it's installed via Package Manager / UPM and shows false warning on OneSignal SDK Setup window.

# Testing
## Unit testing
There are currently no unit tests for editor scripts like InstallEdm4uStep. Moreover it's pretty simple reflection code.

## Manual testing
There should be no warning sign next to Install EDM4U step in OneSignal SDK Setup window when EDM4U is installed using UPM

# Affected code checklist

# Checklist
## Overview
     - If it is hard to explain how any codes changes are related to each other then it most likely needs to be more than one PR

## Testing

## Final pass
      - Simplify with less code, followed by splitting up code into well named functions and variables, followed by adding comments to the code.
      - WIP (Work In Progress) is ok, but explain what is still in progress and what you would like feedback on. Start the PR title with "WIP" to indicate this.


- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/onesignal/onesignal-unity-sdk/554)
